### ISSUE TITLE
Shear inference with Gaussians

### ISSUE THREAD

**Issue by b-remy** [b-remy/gems#5]
I made a first try with Gaussian profiles and odd pixel size without success yet... https://colab.research.google.com/drive/1_O1Li9NunvmXjedZtTf2AmdBFvDd7DZm?usp=sharing
<img width="317" alt="Capture d’écran 2022-02-02 à 19 30 20" src="https://user-images.githubusercontent.com/30293694/152215676-9d665726-6173-45d9-a11a-c34798cea6a8.png">

In this notebook galaxy and psf profiles are fixed, only the shear is free so I expected that the problem was simpler than @EiffL previous experiement (https://colab.research.google.com/drive/1w2W_ISV_7sMYry7VzPFWTWk1b71AcBge?usp=sharing#scrollTo=xnZGlmijI7k-).

So I guess there is something wrong in what I did, maybe there are too much profiles or I should let the hlr free or there is not enough noise

**Comment by b-remy** [comment]
Only shear inference works well eventually, just had to clean the code:
<img width="415" alt="Capture d’écran 2022-02-04 à 17 44 30" src="https://user-images.githubusercontent.com/30293694/152568345-d38a8563-4311-4ff4-b571-686afe180b2f.png">
<img width="614" alt="Capture d’écran 2022-02-04 à 17 45 32" src="https://user-images.githubusercontent.com/30293694/152568364-2b290e43-0f56-43ca-a119-b85086237f2a.png">

next step: 
- [x] enable `tf.function` when sampling
- [x] fit the galaxy profile, intrinsic e and shear.

**Comment by b-remy** [comment]
I am now letting free the shear and the galaxies intrinsic ellipticities (for 25 galaxies) and there is convergence:
<img width="412" alt="Capture d’écran 2022-02-07 à 09 56 09" src="https://user-images.githubusercontent.com/30293694/152756586-9543e4e8-c1ff-40d7-ad12-d31c062a609a.png">
<img width="465" alt="Capture d’écran 2022-02-07 à 09 56 27" src="https://user-images.githubusercontent.com/30293694/152756600-ba5eb00f-6780-4878-86b3-41d0aeffaa51.png">

In this example `gamma2` is biased. I expect this bias to be reduced if I increase the number of galaxies for inference. This is my next experiment. Then I will let the galaxies `hlr` free as well to fit all the Gaussian profiles shape parameters.

**Comment by EiffL** [comment]
Nice :-) Hummmm.... I would expect you would have larger error bars but I don't see why you would get much of a bias on gamma 2.... That's unexpected...

**Comment by b-remy** [comment]
Yes I agree. Then I suspect that it is because I am trying to infer zero shear for `gamma2`. If I remember correctly, the gradients of the current interpolator are not well defined for zero shear, maybe it is inducing a bias... I'll try with non zero shear for both `gamma1` and `gamma2`

**Comment by b-remy** [comment]
Two runs the for inference of shear and galaxies ellipticities, it seems to work well ([19483e0](https://github.com/b-remy/gems/pull/7/commits/19483e0e06e9b65f948d31df16124e4376347bed))
<img width="469" alt="Capture d’écran 2022-02-08 à 14 58 02" src="https://user-images.githubusercontent.com/30293694/153002540-6bce17b9-442e-41e8-b0c9-94b92af2678a.png">
<img width="466" alt="Capture d’écran 2022-02-08 à 14 56 32" src="https://user-images.githubusercontent.com/30293694/153002547-14a4d227-baf1-4204-a23c-05e072ff94e5.png">

Next step: fit also the radius and the flux

**Comment by b-remy** [comment]
Inference of 25 galaxy ellipticities, radius and constant shear ([2e1205d](https://github.com/b-remy/gems/pull/7/commits/2e1205d863535ba7a8c24971b0bf956f5d5615ef)):

<img width="682" alt="Capture d’écran 2022-02-08 à 20 22 15" src="https://user-images.githubusercontent.com/30293694/153061165-36b6c365-76d7-4995-beb6-7b6bb8d22633.png">

<img width="669" alt="Capture d’écran 2022-02-08 à 20 24 18" src="https://user-images.githubusercontent.com/30293694/153061115-e4da64a5-ccf5-4303-8257-7d41f2ed7f2e.png">


<img width="575" alt="Capture d’écran 2022-02-08 à 20 24 24" src="https://user-images.githubusercontent.com/30293694/153061079-22b4c304-06e2-4f10-8a43-99cdcc4ae5d6.png">

So this a good news and what we wanted to demonstrate with this issue!

I think now we can try to see how the shear inference responds when we don't have the same model in the simulations (more complex like sersic profiles) and in the forward model (e.g. gaussians).
